integrations/helm/chartsync/chartsync.go

```diff
@@ -509,7 +509,7 @@ func (chs *ChartChangeSync) shouldUpgrade(chartsRepo string, currRel *hapi_relea
 
 	// Get the desired release state
 	opts := release.InstallOptions{DryRun: true}
-	tempRelName := currRel.GetName() + "-temp"
+	tempRelName := string(fhr.UID)
 	desRel, err := chs.release.Install(chartsRepo, tempRelName, fhr, release.InstallAction, opts, &chs.kubeClient)
 	if err != nil {
 		return false, err
```

Comment:
Maybe append the HelmRelease name to the error

Reply:
Another thing we could do is use the release name from the spec  (`fhr.Spec.ReleaseName`) for informative errors (just like I did with the log entry) but use the given `releaseName` for running operations.

Reply:
sounds good to me